staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -459,9 +466,9 @@ func pollImmediateInternal(wait WaitFunc, condition ConditionFunc) error {
 // Some intervals may be missed if the condition takes too long or the time
 // window is too short.
 func PollInfinite(interval time.Duration, condition ConditionFunc) error {
-	done := make(chan struct{})
-	defer close(done)
-	return PollUntil(interval, condition, done)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	return poll(ctx, false, poller(interval, 0), condition.WithContext())
```

Comment:
Seems like the poller aught to take a context

Reply:
`poller` returns a `WaitFunc`
```
WaitFunc func(done <-chan struct{}) <-chan struct{}
```
- the wait function periodically sends to the returned channel so we can invoke `condition` function at interval. abstracting such a channel with `context` would not look intuitive at all.
-  the `done` channel is used to signal the item generator to quit, not on the path of the user/caller goroutine.